packages/core/src/common/command.ts

```diff
@@ -324,12 +331,17 @@ export class CommandRegistry implements CommandService {
     /**
      * Get a toggled handler for the given command or `undefined`.
      */
-    getToggledHandler(commandId: string): CommandHandler | undefined {
+    // tslint:disable-next-line:no-any
+    getToggledHandler(commandId: string, ...args: any[]): CommandHandler | undefined {
         const handlers = this._handlers[commandId];
         if (handlers) {
             for (const handler of handlers) {
-                if (handler.isToggled) {
-                    return handler;
+                try {
+                    if (handler.isToggled && handler.isToggled(...args)) {
```

Comment:
Of course it is when I merge that I see this...

Should be `!handler.isToggled || handler.isToggled(...args)`.

Reply:
just don't forget to make a follow up

Reply:
Turns out that the logic is fine the way it is, as I didn't know what the toggled state really meant.

See https://github.com/theia-ide/theia/pull/5906#issuecomment-519986808

Reply:
yes, seen it 😄